teps/0089-spire-support.md

```diff
@@ -0,0 +1,184 @@
+---
+status: proposed
+title: SPIRE Support
+creation-date: '2021-10-04'
+last-updated: '2021-11-18'
+authors:
+- '@priyawadhwa'
+---
+
+# TEP-0089: SPIRE Support
+
+<!--
+A table of contents is helpful for quickly jumping to sections of a TEP and for
+highlighting any additional information provided beyond the standard TEP
+template.
+
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Requesting an SVID/Signature from the Controller Image](#requesting-an-svidsignature-from-the-controller-image)
+  - [Enabling SPIRE in Tekton](#enabling-spire-in-tekton)
+- [Test Plan](#test-plan)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+This TEP covers integrating Tekton and Tekton Chains with [SPIFFE/SPIRE](https://spiffe.io/), which would provide a more secure supply chain for Tekton users. It would also guarantee [non-falsifiable provenance](https://slsa.dev/requirements#non-falsifiable), which is a requirement for [SLSA Level 3](https://slsa.dev/levels).
+With this integration, Tekton will be one step closer to SLSA Level 3 compliance.
+
+Currently, Tekton Chains observes Tekton and waits for TaskRuns to complete.
+Once it sees that a TaskRun has completed it tries to sign any artifacts that were built and also generates provenance.
+
+There are a couple issues with this:
+1. Tekton Chains has no way to verify that a TaskRun in the cluster was created by Tekton
+2. Tekton Chains has no way to verify that the TaskRun it received wasn't modified by anybody other than Tekton
+
+This is where SPIRE comes in!
+SPIRE can be used to request signatures and certificates (SVIDs) for a given workload (this workload could be a result, or in our case an entire TaskRun yaml).
+SPIRE can be configured to only issue SVIDs to certain k8s workloads (e.g., you can set up SPIRE to only issue SVIDs to the Tekton controller) with a SPIRE [k8s Workload Attestor](https://github.com/spiffe/spire/blob/main/doc/plugin_agent_workloadattestor_k8s.md).
```

Comment:
We should also add in that we need to do Node attestation before workload attestation can occur. [K8s Node Attestor](https://github.com/spiffe/spire/blob/main/doc/plugin_server_nodeattestor_k8s_sat.md) could be used. If lets say the customer is using TPMs, we can use a [TPM Node Attestor](https://github.com/boxboat/spire-tpm-plugin)